### PR TITLE
add in required callbacks to support validate ddoc checks

### PR DESCRIPTION
**Pull request by garrensmith**

### CODE REVIEW

File: src/ateles.erl [hunk 1]
```diff
@@ -85,6 +88,35 @@ map_docs({CtxId, _} = Ctx, Docs) ->
     end, Docs)}.
 
 
+acquire_context() ->
+    CtxId = <<"garren-test">>,
```
Comment (iilyak) [inline src/ateles.erl]: Does it suppose to be hardcoded?

Comment (garrensmith) [reply src/ateles.erl]: Sorry. I've fixed that now

File: src/ateles.erl [hunk 2]
```diff
@@ -85,6 +88,35 @@ map_docs({CtxId, _} = Ctx, Docs) ->
...
+
+
+release_context(Ctx) ->
+    ateles_server:release(Ctx).
+
+try_compile(Ctx, FunType, FunName, RawFunSrc) ->
+    FunSrc = case ateles_util:rewrite(Ctx, RawFunSrc) of
+        {ok, RewriteSrc} -> 
+            RewriteSrc;
+        {error, {_Status, RewriteResult}} ->
```
Comment (iilyak) [inline src/ateles.erl:104]: How do we handle a case when `ibrowse:send_req_direct` called from https://github.com/cloudant-labs/ateles/blob/master/src/ateles_util.erl#L124 returns `{error, Reason}` see https://github.com/cloudant-labs/ateles/blob/master/src/ateles_util.erl#L135 ?

**Review by iilyak — APPROVED** [review]
+1

**Pull request merged**